frontend/ui/uimanager.lua

```diff
@@ -86,6 +86,25 @@ function UIManager:init()
               self.event_handlers["Suspend"]()
             end
         end
+        if not G_reader_settings:readSetting("ignore_power_sleepcover") then
+            self.event_handlers["SleepCoverClosed"] = function()
+                Device.is_cover_closed = true
+                self.event_handlers["Suspend"]()
+            end
+            self.event_handlers["SleepCoverOpened"] = function()
+                Device.is_cover_closed = false
+                self.event_handlers["Resume"]()
+            end
+        else
+            -- Closing/opening the cover will still wake up the device, so we
+            -- need to put it back to sleep if we are in screen saver mode
+            self.event_handlers["SleepCoverClosed"] = function()
```

Comment:
You will need to use the methods @oldhasbeen mentioned in https://github.com/koreader/koreader/issues/2239 to detect the false sleep.

The rule of thumb here is the kernel will always wake up the device to deliver any input event, so we should not ignore events by removing it from the input_map. Instead, we should capture the event and act accordingly to make sure the device is back to true sleep.